From 574477ab9365f63d40714be298252e12f834e518 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:53 -0400
Subject: [tools] perf/annotate: Reuse path from the result of addr2line

Message-id: <1408462094-14194-125-git-send-email-jolsa@redhat.com>
Patchwork-id: 88103
O-Subject: [PATCH RHEL7.1 BZ1131394 124/325] perf annotate: Reuse path from the result of addr2line
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 89da393c171926d3372f573d752be5ced98038eb
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Wed Sep 11 14:09:26 2013 +0900

description
===========
In the symbol__get_source_line(), path and src_line->path will have same
value, but they were allocated separately, and leaks one.  Just share
path to src_line->path.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/annotate.c b/tools/perf/util/annotate.c
index 7eae548..c6fd187 100644
--- a/tools/perf/util/annotate.c
+++ b/tools/perf/util/annotate.c
@@ -1143,11 +1143,7 @@ static int symbol__get_source_line(struct symbol *sym, struct map *map,
   if (getline(&path, &line_len, fp) < 0 || !line_len)
    goto next_close;
 
-  src_line->path = malloc(sizeof(char) * line_len + 1);
-  if (!src_line->path)
-   goto next_close;
-
-  strcpy(src_line->path, path);
+  src_line->path = path;
   insert_source_line(&tmp_root, src_line);
 
  next_close:
-- 
1.7.1